### ISSUE TITLE
Investigate way to reduce usage friction

### ISSUE THREAD

**Issue by variadico** [variadico/noti#32]
Currently, the biggest obstacle to using `noti` is remembering to use it. I've tried to ease this a little by allowing the command to be run,

* at the beginning, `noti foo`
* at the end, `foo; noti`
* and after running a command, `foo`, remember, `noti -w $(pgrep foo)`.

**Comment by cryptix** [comment]
I've also thought about integrating noti into the shells (bash, zsh, ...), I'm just not sure what kind of hooks they offer.

Basically I'd like noti to grab every command that I started and that runs longer than 1 minute.

**Comment by variadico** [comment]
> Basically I'd like noti to grab every command that I started and that runs longer than 1 minute.

Yeah, I was thinking the exact same thing. I very briefly messed with some ZSH hook that could auto-prepend every command with `noti`. But, then that made the history look like this.

```
noti ls
noti clear
noti brew update
noti git status
```

Which kinda sucks. I need to spend some more time looking into this.

**Comment by tko** [comment]
zsh and I think bash as well have `preexec` and `precmd` hooks you can use like in https://gist.github.com/aehlke/992798

**Comment by ccjensen** [comment]
I've been using the zsh notify "plugin" to get the "notify automatically when command takes longer than x":
https://github.com/marzocchi/zsh-notify

might be worth seeing how it accomplishes that?

**Comment by ctf0** [comment]
i have this in my zsh
```bash
preexec() {
    noti $@
}

autoload -U add-zsh-hook     # Load the zsh hook module.
add-zsh-hook preexec           # Adds the hook
```
